projects/observability/src/pages/apis/service-detail/instrumentation/components/instrumentation-details.component.ts

```diff
@@ -44,4 +66,32 @@ export class InstrumentationDetailsComponent {
   public onClickBack(): void {
     this.router.navigate(['../'], { relativeTo: this.route });
   }
+
+  public getHeaderIcon(score: number): string {
+    if (score < 50) {
+      return IconType.Close;
+    }
+
+    if (score < 70) {
+      return IconType.Warning;
+    }
+
+    return IconType.Checkmark;
+  }
+
+  public getIconColor(score: number): string {
+    return this.serviceInstrumentationService.getColorForScore(score).dark;
+  }
+
+  public getHeaderSummary(heuristicScore: HeuristicScoreInfo): string {
+    if (heuristicScore.score < 0) {
+      return 'Check skipped as no eligible spans in this run';
+    }
+
+    const sampleSize = Number(heuristicScore.sampleSize);
+    const failureCount = Number(heuristicScore.failureCount);
+    const percentFailed = (failureCount / sampleSize) * 100;
+
+    return `${percentFailed}% of ${heuristicScore.sampleType}s failed this check`;
```

Comment:
Do we need `Number` casting here? What are the expected response values?
Asking cause if not, we can do destructing to get all values from `heuristicScore`.

Reply:
Yes, the backend sends these values as strings. Some third-party library issue in the backend that won't let them send integers ([discussion][1]). // @Harnoor-se7en 

[Example][2]

[1]: https://razorpay.slack.com/archives/CU5GKS8MQ/p1662542485001299?thread_ts=1660701815.668959&cid=CU5GKS8MQ
[2]: https://qoi-qs.concierge.stage.razorpay.in/v1/score/vendor-payments-web

Reply:
Just a doubt, what happens if we update the type of `heuristicScore.sampleSize` to have `sampleSize` as `number`?
Does it automatically typecast response? Can you check once?

Reply:
The fixture will start throwing type errors if I change the type to `number`. I can change strings to numbers in the fixture but that wouldn't be a correct representation of the actual backend response. Not sure how to work around this.

![Screenshot 2022-10-06 at 9 35 15 AM](https://user-images.githubusercontent.com/29686866/194211715-b17b0a05-44b1-4815-a538-218e73966f1a.png)